x/fswap/keeper/proposal.go

```diff
@@ -0,0 +1,161 @@
+package keeper
+
+import (
+	"github.com/Finschia/finschia-sdk/store/prefix"
+	sdk "github.com/Finschia/finschia-sdk/types"
+	"github.com/Finschia/finschia-sdk/types/errors"
+	bank "github.com/Finschia/finschia-sdk/x/bank/types"
+	"github.com/Finschia/finschia-sdk/x/fswap/types"
+)
+
+func (k Keeper) MakeSwap(ctx sdk.Context, swap types.Swap, toDenomMetadata bank.Metadata) error {
+	stats, err := k.getSwapStats(ctx)
+	if err != nil {
+		return err
+	}
+	if int(stats.SwapCount) >= k.config.MaxSwaps && !k.isUnlimited() {
```

Comment:
It must be `>` instead of `>=`. And this verification must be relocated to the back. It's because we should allow the updates.

Reply:
You must also relocate the verification. If not, it will allow one to surpass the threshold.